#!/usr/bin/env python
PACKAGE = "usma_triclops"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Camera_System Dynamic Parameters
gen.add("disp_max",     int_t,    0, "disp_max",        70,     1, 240)
gen.add("disp_min",     int_t,    0, "disp_min",        0,      0, 239)
gen.add("disp_map_on",  int_t,    0, "disp_map_on",     0,      0, 1)
gen.add("disp_map_max", int_t,    0, "disp_map_max",    255,    0, 254)
gen.add("disp_map_min", int_t,    0, "disp_map_min",    0,      0, 254)
gen.add("stereo_mask",  int_t,    0, "stereo_mask",     11,     0, 15)

# Line_Filter Dynamic Parameters
gen.add("thresh_val",       int_t,    0, "thresh_val",      243,    0, 255)
gen.add("erosion_size",     int_t,    0, "erosion_size",    5,      0, 25)
gen.add("h_rho",            int_t,    0, "h_rho",           1,      0, 25)
gen.add("h_theta",          int_t,    0, "h_theta",         180,    0, 360)
gen.add("h_thresh",         int_t,    0, "h_thresh",        30,     0, 255)
gen.add("h_minLineLen",     int_t,    0, "h_minLineLen",    20,     0, 250)
gen.add("h_maxLineGap",     int_t,    0, "h_maxLineGap",    7,      0, 250)


gen.add("int_param",    int_t,    0, "An Integer parameter", 50,  0, 100)
gen.add("double_param", double_t, 0, "A double parameter",    .5, 0,   1)
gen.add("str_param",    str_t,    0, "A string parameter",  "Hello World")
gen.add("bool_param",   bool_t,   0, "A Boolean parameter",  True)

size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),
                       gen.const("Medium",     int_t, 1, "A medium constant"),
                       gen.const("Large",      int_t, 2, "A large constant"),
                       gen.const("ExtraLarge", int_t, 3, "An extra large constant")],
                     "An enum to set size")

gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "dynamic_tutorials", "Tutorials"))
